--- origsrc/kdenlive-0.8.2.1/src/CMakeLists.txt	2011-12-07 02:02:13.000000000 -0600
+++ src/kdenlive-0.8.2.1/src/CMakeLists.txt	2011-12-19 04:21:29.241336800 -0600
@@ -283,24 +283,24 @@ if(CMAKE_SYSTEM_NAME STREQUAL "GNU/kFree
   add_definitions(-DQ_OS_KFREEBSD)
 endif(CMAKE_SYSTEM_NAME STREQUAL "GNU/kFreeBSD")
 
-if(APPLE OR CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" OR CMAKE_SYSTEM_NAME STREQUAL "NetBSD" OR CMAKE_SYSTEM_NAME STREQUAL "GNU/kFreeBSD" OR NO_JOGSHUTTLE)
+if(NOT CMAKE_SYSTEM_NAME STREQUAL "Linux" OR NO_JOGSHUTTLE)
   add_definitions(-DNO_JOGSHUTTLE)
-else(APPLE OR CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" OR CMAKE_SYSTEM_NAME STREQUAL "NetBSD" OR CMAKE_SYSTEM_NAME STREQUAL "GNU/kFreeBSD" OR NO_JOGSHUTTLE)
+else(NOT CMAKE_SYSTEM_NAME STREQUAL "Linux" OR NO_JOGSHUTTLE)
   set(kdenlive_SRCS jogshuttle.cpp jogaction.cpp jogshuttleconfig.cpp ${kdenlive_SRCS})
-endif(APPLE OR CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" OR CMAKE_SYSTEM_NAME STREQUAL "NetBSD" OR CMAKE_SYSTEM_NAME STREQUAL "GNU/kFreeBSD" OR NO_JOGSHUTTLE)
+endif(NOT CMAKE_SYSTEM_NAME STREQUAL "Linux" OR NO_JOGSHUTTLE)
 
 if(APPLE OR OPENGL_FOUND)
   add_definitions(-DUSE_OPEN_GL)
   set(kdenlive_SRCS videoglwidget.cpp ${kdenlive_SRCS})
 endif(APPLE OR OPENGL_FOUND)
 
-if(NOT APPLE AND NOT CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" AND NOT CMAKE_SYSTEM_NAME STREQUAL "GNU/kFreeBSD")
+if(CMAKE_SYSTEM_NAME STREQUAL "Linux")
   set(kdenlive_SRCS
     ${kdenlive_SRCS}
     v4l/v4lcapture.cpp
     v4l/src.c
   )
-endif(NOT APPLE AND NOT CMAKE_SYSTEM_NAME STREQUAL "FreeBSD" AND NOT CMAKE_SYSTEM_NAME STREQUAL "GNU/kFreeBSD")
+endif(CMAKE_SYSTEM_NAME STREQUAL "Linux")
 
 if(QJSON_FOUND)
   set(kdenlive_SRCS
--- origsrc/kdenlive-0.8.2.1/src/kdenlivesettingsdialog.cpp	2011-12-07 02:02:13.000000000 -0600
+++ src/kdenlive-0.8.2.1/src/kdenlivesettingsdialog.cpp	2011-12-19 04:16:32.696375400 -0600
@@ -19,7 +19,7 @@
 
 #include "kdenlivesettingsdialog.h"
 #include "profilesdialog.h"
-#if !defined(Q_OS_FREEBSD) && !defined(Q_OS_KFREEBSD)
+#if defined(Q_OS_LINUX)
 #include "v4l/v4lcapture.h"
 #endif
 #include "blackmagic/devices.h"
@@ -97,7 +97,7 @@ KdenliveSettingsDialog::KdenliveSettings
     QWidget *p4 = new QWidget;
     m_configCapture.setupUi(p4);
 
-#if !defined(Q_WS_MAC) && !defined(Q_OS_FREEBSD) && !defined(Q_OS_KFREEBSD)
+#ifdef Q_OS_LINUX
 
     // Video 4 Linux device detection
     for (int i = 0; i < 10; i++) {
--- origsrc/kdenlive-0.8.2.1/src/stopmotion/stopmotion.cpp	2011-12-07 02:02:13.000000000 -0600
+++ src/kdenlive-0.8.2.1/src/stopmotion/stopmotion.cpp	2011-12-19 04:18:26.171865800 -0600
@@ -17,7 +17,7 @@
 
 #include "stopmotion.h"
 #include "../blackmagic/devices.h"
-#if !defined(Q_OS_FREEBSD) && !defined(Q_OS_KFREEBSD)
+#ifdef Q_OS_LINUX
 #include "../v4l/v4lcapture.h"
 #endif
 #include "../slideshowclip.h"
@@ -257,7 +257,7 @@ StopmotionWidget::StopmotionWidget(Monit
         // Found a BlackMagic device
     }
     if (QFile::exists(KdenliveSettings::video4vdevice())) {
-#if !defined(Q_WS_MAC) && !defined(Q_OS_FREEBSD) && !defined(Q_OS_KFREEBSD)
+#ifdef Q_OS_LINUX
         // Video 4 Linux device detection
         for (int i = 0; i < 10; i++) {
             QString path = "/dev/video" + QString::number(i);
@@ -400,7 +400,7 @@ void StopmotionWidget::slotUpdateDeviceH
     }
     m_layout->removeWidget(m_frame_preview);
     if (data == "v4l") {
-#if !defined(Q_WS_MAC) && !defined(Q_OS_FREEBSD) && !defined(Q_OS_KFREEBSD)
+#ifdef Q_OS_LINUX
         m_bmCapture = new V4lCaptureHandler(m_layout);
         m_bmCapture->setDevice(capture_device->itemData(capture_device->currentIndex(), Qt::UserRole + 1).toString(), capture_device->itemData(capture_device->currentIndex(), Qt::UserRole + 2).toString());
 #endif
--- origsrc/kdenlive-0.8.2.1/src/wizard.cpp	2011-12-07 02:02:13.000000000 -0600
+++ src/kdenlive-0.8.2.1/src/wizard.cpp	2011-12-19 04:19:21.853050600 -0600
@@ -20,7 +20,7 @@
 #include "wizard.h"
 #include "kdenlivesettings.h"
 #include "profilesdialog.h"
-#if !defined(Q_OS_FREEBSD) && !defined(Q_OS_KFREEBSD)
+#ifdef Q_OS_LINUX
 #include "v4l/v4lcapture.h"
 #endif
 #include "kdenlive-config.h"
@@ -158,7 +158,7 @@ Wizard::Wizard(bool upgrade, QWidget *pa
 
 void Wizard::slotDetectWebcam()
 {
-#if !defined(Q_WS_MAC) && !defined(Q_OS_FREEBSD) && !defined(Q_OS_KFREEBSD)
+#ifdef Q_OS_LINUX
     m_capture.v4l_devices->blockSignals(true);
     m_capture.v4l_devices->clear();
 
